prod/westeurope/internal/api/functions_admin/function_app_slot_staging/terragrunt.hcl

```diff
@@ -0,0 +1,165 @@
+
+
+# Include all settings from the root terragrunt.hcl file
+include {
+  path = find_in_parent_folders()
+}
+
+
+terraform {
+  source = "git::git@github.com:pagopa/io-infrastructure-modules-new.git//azurerm_function_app_slot?ref=v2.0.33"
+}
+
+inputs = {
+  name                       = "staging"
+  resource_group_name        = dependency.resource_group.outputs.resource_name
+  function_app_name          = dependency.function_app.outputs.name
+  function_app_resource_name = dependency.function_app.outputs.resource_name
+  app_service_plan_id        = dependency.function_app.outputs.app_service_plan_id
+  storage_account_name       = dependency.function_app.outputs.storage_account.name
+  storage_account_access_key = dependency.function_app.outputs.storage_account.primary_access_key
+
+  runtime_version = "~3"
+
+  auto_swap_slot_name = "production"
+
+  application_insights_instrumentation_key = dependency.application_insights.outputs.instrumentation_key
+
+  # site_config
+  pre_warmed_instance_count = 1
+
+  app_settings = {
+    FUNCTIONS_WORKER_RUNTIME     = "node"
+    WEBSITE_NODE_DEFAULT_VERSION = "10.14.1"
+    WEBSITE_RUN_FROM_PACKAGE     = "1"
+    NODE_ENV                     = "production"
+
+    COSMOSDB_URI               = dependency.cosmosdb_account.outputs.endpoint
+    COSMOSDB_KEY               = dependency.cosmosdb_account.outputs.primary_master_key
+    COSMOSDB_NAME              = dependency.cosmosdb_database.outputs.name
+    COSMOSDB_CONNECTION_STRING = dependency.cosmosdb_account.outputs.connection_strings[0]
+
+    StorageConnection = dependency.storage_account.outputs.primary_connection_string
+
+    AssetsStorageConnection = dependency.storage_account_assets.outputs.primary_connection_string
+
+    AZURE_APIM                = "io-p-apim-api"
+    AZURE_APIM_HOST           = "api-internal.io.italia.it"
+    AZURE_APIM_RESOURCE_GROUP = "io-p-rg-internal"
+
+    MESSAGE_CONTAINER_NAME = dependency.storage_container_message-content.outputs.name
+
+    UserDataArchiveStorageConnection = dependency.storage_account_user-data-download.outputs.primary_connection_string
+    USER_DATA_CONTAINER_NAME         = dependency.storage_container_user-data-download.outputs.name
+
+    PUBLIC_API_URL           = "http://api-internal.io.italia.it/"
+    PUBLIC_DOWNLOAD_BASE_URL = "https://${dependency.storage_account_user-data-download.outputs.primary_blob_host}/${dependency.storage_container_user-data-download.outputs.name}"
+
+    SESSION_API_URL                 = "https://${dependency.app_service_appbackend.outputs.default_site_hostname}"
+    UserDataBackupStorageConnection = dependency.storage_account_userbackups.outputs.primary_connection_string
+    USER_DATA_BACKUP_CONTAINER_NAME = dependency.storage_container_user-data-backup.outputs.name
+    USER_DATA_DELETE_DELAY_DAYS     = 7
+
+    // Disabled functions
+    "AzureWebJobs.UserDataProcessingTrigger.Disabled"  = "1"
+    "AzureWebJobs.UpdateVisibleServicesCache.Disabled" = "1"
```

Comment:
we must mark someway these settings as "slot settings" otherwise they will leak into production when swapping. moreover in every slot (staging and production) we must add a slot setting:

SLOT_TASK_HUBNAME=StagingTaskHub (for staging)
SLOT_TASK_HUBNAME=ProductionTaskHub (for production)

Reply:
I've added everywhere required the **SLOT_TASK_HUBNAME** setting with its proper key.
Unfortunately there is no possibility in app_settings to set the "Deployment slot setting" flag as per arm template definition.
So far, I've just put a comment before each "AzureWebJobs.*.Disabled"  = "1" part. We need to set the flag manually from the portal.